src/app/producer-shelf/components/producer-shelf/producer-shelf.component.ts

```diff
@@ -47,14 +78,40 @@ export class ProducerShelfComponent implements OnInit, OnDestroy {
       .then(actionsObservable => actionsObservable.subscribe(this.onActionsChanged.bind(this)))
       .then(actionsSubscription => (this.actionsSubscription = actionsSubscription))
       .catch(error => this.logger.data(error).error('Failed subscribing to actions.'))
+
+    this.configurationService.getShelfConfiguration().subscribe(shelf => {
+      this.shelfConfiguration = shelf
+      this.updateActionPanels()
+    })
```

Comment:
I try to follow a practice witch all the time we unsubscribe for the observables. I know that will emit only one time, but we still can:

```
private readonly unsubscribe$: Subject<null> = new Subject<null>()

this.configurationService.getShelfConfiguration().pipe(takeUntil(this.unsubscribe$)).subscribe(shelf => {
    this.shelfConfiguration = shelf
    this.updateActionPanels()
  })
  
   public ngOnDestroy(): void {
    this.unsubscribe$.unsubscribe()
  }
```

Is that a good practice, or is overkill?

Reply:
I think it is good.
It makes it so we don't need to store _n_ subscriptions and unsubscribe all of them, but instead we just store the single `unsubscribe` and subscribe on that one instead.